src/app/app.component.html

```diff
@@ -62,46 +62,13 @@
 			<button mat-raised-button color="primary" [disabled]="!addPackage.value || addPackage.invalid">Add</button>
 		</form>
 
-		<ng-container *ngIf="apiDatas$ | async as apiDatas">
-			<mat-selection-list
-				*ngIf="apiDatas.length; else noPackageNames"
-				[formControl]="selectedPackageNames"
-				#selectionList
-			>
-				<div class="option-wrapper" *ngFor="let apiData of apiDatas; trackBy: trackByPackageName">
-					<mat-list-option [value]="apiData.packageName">
-						<span class="package-list-option">
-							<span>{{ apiData.packageName }}</span>
-							<span>
-								<mat-icon>download</mat-icon>
-								{{ apiData.total | number }}
-							</span>
-						</span>
-					</mat-list-option>
-
-					<button
-						class="button-wrapper"
-						(click)="removePackageName(apiData.packageName)"
-						(keydown)="handleSelectAsClick($event)"
-					>
-						<button mat-icon-button color="accent" aria-label="Remove npm package" tabindex="-1">
-							<mat-icon class="delete-icon">delete</mat-icon>
-						</button>
-					</button>
-
-					<a
-						class="button-wrapper"
-						[href]="'https://www.npmjs.com/package/' + apiData.packageName"
-						target="_blank"
-						(keydown)="handleSelectAsClick($event)"
-					>
-						<button mat-icon-button color="accent" aria-label="Remove npm package" tabindex="-1">
-							<mat-icon class="new-tab-icon">open_in_new</mat-icon>
-						</button>
-					</a>
-				</div>
-			</mat-selection-list>
-		</ng-container>
+		<!-- list of selected libraries -->
+		<app-package-list
+			*ngIf="apiDatas$ | async as apiDatas"
```

Comment:
Not sure if this `*ngIf` is still needed with refactoring out to its own component.
I would put
```
[registryData]="apiDatas$ | async"
```
and then deal with the changes in that component's `onChanges` method

Reply:
I think in this case it is fine to just pass the data `[registryData]="apiDatas$ | async"` into the component. It is a dumb one, so no need to even implement onChange. 
But with onChange, maybe Mark can say more to it, but we want to avoid using it. If you have to implement onChange interface it means you are kinda losing the reactivity through your application and have to 'manually' react on input changes